jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -81,6 +81,12 @@ def parse_data(app, file_obj, ext=None, data_label=None,
         `~astropy.utils.data.download_file` or
         `~astroquery.mast.Conf.timeout`).
     """
+    file_obj = get_cloud_fits(file_obj)
+
+    if not isinstance(file_obj, str):
+        _parse_image(app, file_obj, data_label, ext=ext, parent=parent)
+        return
+
```

Comment:
The [errors in CI](https://github.com/spacetelescope/jdaviz/actions/runs/13812067649/job/38635748751#step:5:9087) come from this block's placement. Any file path or object will go into `get_cloud_fits` at the moment, but `get_cloud_fits` isn't prepared to work on inputs that aren't strings. 

You might be able to fix this by moving this chunk of code under the following if block (`if isinstance(file_obj, str):`), so only strings see this logic.